### DISCUSSION TITLE
Get data statistics from websocket client

### DISCUSSION THREAD

**Discussion by willmcgugan** [aio-libs/aiohttp#7572] [Q&A] [answered]
Hi,

I would like to collect usage statistics from a websocket client connection. i.e. number of bytes sent and number of bytes received. Ideally the number of bytes on the wire, but uncompressed would also work.

What is the easiest way to collect these stats? I don't see a public interface.

Thanks.

**Comment by Dreamsorcerer** [comment, marked as answer]
I don't see an easy option with the current code. It could maybe be possible by using a custom response class, though the actual sending is handled by the writer class (passed into the response class), which is hardcoded: https://github.com/aio-libs/aiohttp/blob/675579699422680607108a7dd68c85ec5284220c/aiohttp/client.py#L890. I suppose you could always monkeypatch that object in your custom response class.

I think there has been requests to introduce tracing to websockets, so if you wanted to try and expand the current tracing stuff to websockets, maybe that would be worth looking at (I've not personally done anything with the tracing API before, so not sure how well it matches up).

By the way, your Pygame book was a big influence on me learning to program ~15 years ago.

**Reply by willmcgugan** [reply]
Thanks.

> By the way, your Pygame book was a big influence on me learning to program ~15 years ago.

Has it really been that long?! :-)

**Reply by Dreamsorcerer** [reply]
Apparently, I was surprised as well. :P
